docs/fobnail-data-format.md

```diff
@@ -0,0 +1,87 @@
+# Fobnail data format
+
+Fobnail uses CBOR (with a few exceptions) for transmitting structured data both
+from Fobnail to Attester and from Attester to Fobnail. We use subset of
+[RFC7049](https://datatracker.ietf.org/doc/html/rfc7049) (no tagging and no
+extensions).
+
+## EK certificate
+
+EK certificate is transferred as a raw DER. We don't add any more data so there
+is no need to wrap it into CBOR.
+
+## AIK
+
+AIK is transferred in TPM format (TPM2B_PUBLIC structure). Re-encoding this into
+CBOR would strip essential data. Fobnail needs unmodified TPM2B_PUBLIC to
+successfully perform the "make credential" operation which is needed for
+credential activation.
+
+## Metadata
+
+Metadata is sent in CBOR-encoded format (also described
+[here](https://github.com/fobnail/docs/blob/main/dev-notes/metadata.md) in more
+detail). Example metadata looks like this (for readability, presented as JSON)
+
+```json
+{
+    // metadata version, used to detect backward-incompatible changes
+    "version": 1,
```

Comment:
@ arturkow2000 can we describe each key that is part of this structure?

Reply:
:+1: also CBOR major type of each field would be good. It would be nice to also mention where this data comes from, but perhaps this should be in `metadata.md` instead.

Reply:
Added brief description of each field including major types.
Details (like how to determine primary network adapter) are described in metadata.md which is linked here